tachikoma-test-client/src/main/kotlin/SendTemplateEmail.kt

```diff
@@ -1,31 +1,31 @@
 import com.google.protobuf.Struct
 import com.google.protobuf.Value
 import com.google.protobuf.util.JsonFormat
+import com.linecorp.armeria.client.Clients
+import com.sourceforgery.jersey.uribuilder.ensureGproto
+import com.sourceforgery.jersey.uribuilder.withoutPassword
 import com.sourceforgery.tachikoma.grpc.frontend.NamedEmailAddress
 import com.sourceforgery.tachikoma.grpc.frontend.maildelivery.EmailRecipient
 import com.sourceforgery.tachikoma.grpc.frontend.maildelivery.MailDeliveryServiceGrpc
 import com.sourceforgery.tachikoma.grpc.frontend.maildelivery.OutgoingEmail
 import com.sourceforgery.tachikoma.grpc.frontend.maildelivery.TemplateBody
 import com.sourceforgery.tachikoma.grpc.frontend.maildelivery.TemplateEngine
-import io.grpc.ManagedChannelBuilder
-import io.grpc.Metadata
 import io.grpc.StatusRuntimeException
-import io.grpc.stub.MetadataUtils
+import java.net.URI
+import java.time.Duration
 import java.time.Instant
 
-private val APITOKEN_HEADER = Metadata.Key.of("x-apitoken", Metadata.ASCII_STRING_MARSHALLER)
-
 fun main(args: Array<String>) {
-    val metadataAuth = Metadata()
-    metadataAuth.put(APITOKEN_HEADER, System.getenv("FRONTEND_API_TOKEN")!!)
-
-    @Suppress("DEPRECATION")
-    val channel = ManagedChannelBuilder.forAddress("localhost", 8070)
-        .usePlaintext()
-        .intercept(MetadataUtils.newAttachHeadersInterceptor(metadataAuth))
-        .build()
+    val frontendUri = URI.create(
+        System.getenv("TACHI_FRONTEND_URI")
+            ?: error("Need to specify env TACHI_FRONTEND_URI")
+    )
 
-    val stub = MailDeliveryServiceGrpc.newBlockingStub(channel)
+    val stub = Clients.builder(frontendUri.withoutPassword().ensureGproto())
+        .addHeader("x-apitoken", frontendUri.userInfo)
```

Comment:
Maybe extract frontendUri.userInfo to val token. 
Just for clarity

Reply:
No strong opinion, so sure.